src/hooks/useBottomSheetDynamicSnapPoints.ts

```diff
@@ -29,7 +29,7 @@ export const useBottomSheetDynamicSnapPoints = (
       animatedHandleHeight.value === INITIAL_HANDLE_HEIGHT ||
       animatedContentHeight.value === 0
     ) {
-      return [INITIAL_SNAP_POINT];
+      return initialSnapPoints.map(_ => INITIAL_SNAP_POINT);
```

Comment:
i would suggested to re-write it this way

```tsx
      return initialSnapPoints.map(() => INITIAL_SNAP_POINT);
```

Reply:
> i would suggested to re-write it this way
> 
> ```tsx
>       return initialSnapPoints.map(() => INITIAL_SNAP_POINT);
> ```

I'm working on another implementation using `Symbol()` as `INITIAL_DYNAMIC_SNAP_POINTS` to pass to `snapPoints` parameter. It should solve #586 #578.

Reply:
@gorhom Or should I abandon this work in preference of this PR?

Reply:
@likern i have resolved #578 with https://github.com/gorhom/react-native-bottom-sheet/commit/95ea72a459f96d40ad583c5579cc72f0e128e5dd, and #586 should be resolved with this pr.

I tried to implement it using `Symbol` as initial values, but it will require a lot of refactoring to internal functionality.

Reply:
this pr is needed when using `useBottomSheetDynamicSnapPoints`, i was returning only one value `INITIAL_DYNAMIC_SNAP_POINTS` but internally i was checking for the snap point with provided index. which will cause the crash.

Reply:
@gorhom Yes, of course I'm talking about correct PR which will take into account this.

Reply:
Something like this.
![Снимок экрана от 2021-08-18 00-18-55](https://user-images.githubusercontent.com/842306/129802076-586ece9e-a4cc-4e2c-9a07-b472a3324ee0.png)

Reply:
@gorhom If 🆗 I will continue implementation. In that case there will be no confusion and accident errors between 
`[-999]` special array and normal array for snap points. No confustion in parameter validations (don't allow negative values - now I get exception because first is rendered `-999` value). It's kind of a hacks which are layering on top of each other.

I think best way is to use symbols to distinguish from normal arrays.

Reply:
@likern what will be the different from this approach ? you'll still need to handle all places `_providedIndex` is referenced to handle one item array `_providedSnapPoints` & `animatedSnapPoints`. for me this approach is easier and correct my mistake on the initial commit for this hook.

i would like to hear your opinion :)

Reply:
@likern with the latest commit i pushed, the `snapPoints` will need to be `> 0 value` or an initial snap point `-999`.

>No confustion in parameter validations (don't allow negative values - now I get exception because first is rendered -999 value). It's kind of a hacks which are layering on top of each other.

I have tested this locally on dynamic snap point example. it should not throw an exception.

Reply:
For getting unique symbol (I mean ES6 Symbol type) user need

- Call `useBottomSheetDynamicSnapPoints` which will return that unique symbol 
It's not possible to accidentally pass `[-999]` special array to `snapPoints` parameter. I think it's more evident what's happening and why.

- When logging and debugging it's difficult to understand what is `[-999]` and what's going on. I initially thought it's a bug, untill I dig into code.
- While now that approach might work, I'm sure there will be more bugs. Because `[-999]` and user's array are just arrays with _valid values_ (from JS perspective) and it's difficult to distinguish them. While array and symbol are completely different (and unique, we have to use only one `INITIAL_DYNAMIC_SNAP_POINTS`, while can spread out `[-999]` directly)
- TypeScript supports `unique symbol` type allow differentiate easily (as in runtime in JS). Now we need some hacks - for example check that first value from array is `-999`. It feels like a hack and that information will be lost later while refactoring and changing code base.

Reply:
It doesn't solve problem that we need to check `animatedSnapPoints` in every place where we use it. I agree with that.

 - But you can't accidentally call `slice()` on `symbol | number[]`, while now you can. And there is some weird calculation (and probably wrong)

Also this code is correct and totally valid, except this weird `[-999]` special handle.
```ts
        invariant(
          index >= -1 && index <= snapPoints.length - 1,
          `'index' was provided but out of the provided snap points range! expected value to be between -1, ${
            snapPoints.length - 1
          }`
        );
```

Reply:
@gorhom I'm OK with any solution that's working ;) Thanks for fast fixing. Will check tomorrow.